htdocs/core/ajax/selectsearchbox.php

```diff
@@ -56,97 +56,97 @@
 
 if (! empty($conf->adherent->enabled) && empty($conf->global->MAIN_SEARCHFORM_ADHERENT_DISABLED) && $user->rights->adherent->lire)
 {
-	$arrayresult['searchintomember']=array('position'=>8, 'shortcut'=>'M', 'img'=>'object_user', 'label'=>$langs->trans("SearchIntoMembers", $search_boxvalue), 'text'=>img_picto('', 'object_user').' '.$langs->trans("SearchIntoMembers", $search_boxvalue), 'url'=>DOL_URL_ROOT.'/adherents/list.php'.($search_boxvalue?'?sall='.urlencode($search_boxvalue):''));
+	$arrayresult['searchintomember']=array('position'=>8, 'shortcut'=>'M', 'img'=>'object_user', 'label'=>$langs->trans("SearchIntoMembers", $search_boxvalue), 'text'=>img_picto('', 'object_user').' '.$langs->trans("SearchIntoMembers", $search_boxvalue), 'url'=>dol_buildpath('/adherents/list.php', 1).($search_boxvalue?'?sall='.urlencode($search_boxvalue):''));
```

Comment:
dol_buildpath() must be used for external module only. If we used dol_buildpath here, we should use it also at everyplace we use the link to same page to have a behavious consistent.

If goal is to have the search done on another page than the official page (because a module want to change this with its own search page), it is better to disable the search entry (by setting MAIN_SEARCHFORM_ADHERENT_DISABLED) and have the module adding its own search entry.

Reply:
at the moment, it is inconsistent because links to list in menus use dol_buildpath and not search forms, so it leads to misunderstanding of the users.
ie for a customed product list